tests/functional/adapter/test_python_model.py

```diff
@@ -65,5 +67,47 @@ def test_changing_schema(self,project):
         write_file(models__simple_python_model_v2, project.project_root + '/models', "simple_python_model.py")
         run_dbt(["run"])
 
+
+m_1 = """
+{{config(materialized='table')}}
+select 1 as id union all
+select 2 as id union all
+select 3 as id union all
+select 4 as id union all
+select 5 as id
+"""
+
+incremental_python = """
+def model(dbt, session):
+    dbt.config(materialized="incremental", unique_key='id')
+    df = dbt.ref("m_1")
+    if dbt.is_incremental:
+        # incremental runs should only apply to 
+        df = df.filter(df.id >= session.sql(f"select max(id) from {dbt.this}").collect()[0][0])
```

Comment:
@jtcohen6  I couldn't figure out a better way to do this with a dataframe. Probably something that we can surface with the snowpark team

Reply:
I don't really like this but also couldn't figure out a way to get around it for now even with `max`. I feel okay leaving it like this for now